packages/use-contractkit/src/connectors/connectors.ts

```diff
@@ -180,14 +207,17 @@ export class InjectedConnector implements Connector {
     defaultType: WalletTypes = WalletTypes.Injected
   ) {
     this.type = defaultType;
-    localStorage.setItem(localStorageKeys.lastUsedWalletType, defaultType);
+    this.kit = newKit(network.rpcUrl);
+    this.network = network;
+  }
+
+  persist() {
```

Comment:
is this persist method the same everywhere? 

it seems like it. 

I recommend writing once and then binding like `updateFeeCurrency` is

Reply:
Yeah I was thinking of that. I liked having the flexibility but could def. abstract it.